custom/covid/management/commands/fetch_deleted_cases.py

```diff
@@ -0,0 +1,87 @@
+import csv
+from casexml.apps.case.xform import get_case_updates
+from datetime import datetime
+from django.core.management.base import BaseCommand
+
+
+from corehq.apps.es import FormES
+from corehq.form_processor.models.cases import CommCareCase, XFormInstance
+from corehq.util.argparse_types import date_type
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--domains', nargs="*",
+            help='Domains to check, will include enterprise-controlled child domains.'
+        )
+        parser.add_argument('--start', type=date_type, help='Start date (inclusive)')
+        parser.add_argument('--end', type=date_type, help='End date (inclusive)')
+
+    def handle(self, **options):
+        filename = "deleted_cases_pull_{}.csv".format(datetime.utcnow().strftime("%Y-%m-%d_%H.%M.%S"))
+        domains = options['domains']
+
+        print(f"Outputting to: {filename}...")
+        with open(filename, 'w', encoding='utf-8') as csv_file:
+            field_names = ['domain', 'case_id', 'case_type', 'date_of_deletion']
+            csv_writer = csv.DictWriter(csv_file, field_names)
+            csv_writer.writeheader()
+            for domain in domains:
+                deleted_case_ids = CommCareCase.objects.get_deleted_case_ids_in_domain(domain)
+                deleted_cases = CommCareCase.objects.get_cases(deleted_case_ids)
```

Comment:
You might wanna chunk this call like here, since this presumably could be a very large number of cases

https://github.com/dimagi/commcare-hq/blob/master/corehq/messaging/management/commands/run_rule.py#L35-L43

Reply:
actually that example is a little complicated, I think all you'd need to do is:
```python
for ids_chunk in deleted_case_ids:
    for case in CommCareCase.objects.get_cases(ids_chunk):
        ...
```